### PR TITLE
Fixed control flags in the test files of GBODE

### PR DESCRIPTION
**Pull request by bernhardbachmann**
Original commit had still the old flag settings of -gbctrl for GBODE

### CODE REVIEW

File: OMCompiler/SimulationRuntime/c/simulation/solver/gbode_util.c [hunk 1]
```diff
@@ -290,18 +346,25 @@ double error_interpolation_gb(DATA_GBODE* gbData, int nIdx, int* idx, double tol
...
+                            (gbData->timeLeft + gbData->timeRight)/2, gbData->y1,
+                            nIdx, idx);
+    // hermite_interpolation_b(gbData->timeLeft,  gbData->yLeft,
+    //                         gbData->timeRight, gbData->yRight, gbData->kRight,
+    //                         (gbData->timeLeft + gbData->timeRight)/2, gbData->y1,
+    //                         nIdx, idx);
+    // linear_interpolation(gbData->timeLeft,  gbData->yLeft,
+    //                     gbData->timeRight, gbData->yRight,
+    //                     (gbData->timeLeft + gbData->timeRight)/2, gbData->y1,
+    //                     nIdx, idx);
```
Comment (AnHeuermann) [inline OMCompiler/SimulationRuntime/c/simulation/solver/gbode_util.c:366]: Remove commented stuff?

```suggestion
```

File: OMCompiler/SimulationRuntime/c/simulation/solver/gbode_main.c [hunk 2]
```diff
@@ -1713,11 +1729,20 @@ int gbode_singlerate(DATA *data, threadData_t *threadData, SOLVER_INFO *solverIn
...
+      if (gbData->ctrl_method != GB_CTRL_CNST && ((gbData->interpolation == GB_INTERPOL_HERMITE_ERRCTRL)  || (gbData->interpolation == GB_DENSE_OUTPUT_ERRCTRL))) {
+        gbData->errValues[0] = fmax(err, gbData->err_int);
+        gbData->stepSize = gbData->lastStepSize * IController(gbData->errValues, gbData->stepSizeValues, gbData->tableau->error_order);
+      }
       // reject step, if interpolaton error is too large
       if ((gbData->err_int > 1 ) && gbData->ctrl_method != GB_CTRL_CNST &&
           ((gbData->interpolation == GB_INTERPOL_HERMITE_ERRCTRL)  || (gbData->interpolation == GB_DENSE_OUTPUT_ERRCTRL))) {
         err = 100;
-        gbData->stepSize = gbData->lastStepSize*IController(&(gbData->err_int), &(gbData->lastStepSize), 1);
+        // gbData->stepSize = gbData->lastStepSize*IController(&(gbData->err_int), &(gbData->lastStepSize), 1);
```
Comment (AnHeuermann) [inline OMCompiler/SimulationRuntime/c/simulation/solver/gbode_main.c:1740]: Remove

```suggestion
```

**Review by AnHeuermann — CHANGES_REQUESTED** [review]
Mostly nitpicking that can be ignored.

Clarification on `rejectedType` would be great. I have no idea what that is.
`sprintf` should also be replaced by `snprintf` before crashing in a log, so we don't need to search for it later.

File: OMCompiler/SimulationRuntime/c/simulation/solver/gbode_main.c [hunk 3]
```diff
@@ -271,6 +272,8 @@ int gbodef_allocateData(DATA *data, threadData_t *threadData, DATA_GBODE *gbData
     char filename[4096];
     sprintf(filename, "%s_ActiveStates.txt", data->modelData->modelFilePrefix);
     gbfData->fastStatesDebugFile = omc_fopen(filename, "w");
+    warningStreamPrint(LOG_STDOUT, 0, "LOG_GBODE_STATES sets -noEquidistantTimeGrid for emitting results!");
```
[the remark this answers is not in the document]
Comment (bernhardbachmann) [reply OMCompiler/SimulationRuntime/c/simulation/solver/gbode_main.c:275]: Yes!
The activity diagram only works properly, if the equidistant time grid interpolation is disabled

File: OMCompiler/SimulationRuntime/c/simulation/solver/gbode_main.c [hunk 4]
```diff
@@ -621,7 +625,7 @@ void gbodef_init(DATA* data, threadData_t* threadData, SOLVER_INFO* solverInfo)
   gbfData->didEventStep = FALSE;
 
   gbfData->time = gbData->time;
-  gbfData->stepSize = gbData->lastStepSize/2.5;
+  gbfData->stepSize = 0.1*gbData->stepSize*IController(&(gbData->err_fast), &(gbData->stepSize), 1);
```
[the remark this answers is not in the document]
Comment (bernhardbachmann) [reply OMCompiler/SimulationRuntime/c/simulation/solver/gbode_main.c:628]: Only the IController can be used here. The other controller need more than one error value.

File: OMCompiler/SimulationRuntime/c/simulation/solver/gbode_util.c [hunk 5]
```diff
@@ -565,13 +628,14 @@ void printSparseJacobianLocal(ANALYTIC_JACOBIAN* jacobian, const char* name) {
 /**
  * @brief Write information on the active fast states on file (activity diagram)
  *
- * @param gbData  Pointer to generik GBODE data struct.
- * @param event   If an event has happend, write zeros else ones
- * @param time    Actual time of reporting
+ * @param gbData       Pointer to generik GBODE data struct.
+ * @param event        If an event has happend, write zeros else ones
+ * @param time         Actual time of reporting
+ * @param rejectedType Type of rejection
```
[the remark this answers is not in the document]
Comment (bernhardbachmann) [reply OMCompiler/SimulationRuntime/c/simulation/solver/gbode_util.c:626]: Explanation has been added to the source code

File: OMCompiler/SimulationRuntime/c/simulation/solver/gbode_util.c [hunk 6]
```diff
@@ -565,13 +628,14 @@ void printSparseJacobianLocal(ANALYTIC_JACOBIAN* jacobian, const char* name) {
...
  */
-void dumpFastStates_gb(DATA_GBODE* gbData, modelica_boolean event, double time) {
+void dumpFastStates_gb(DATA_GBODE* gbData, modelica_boolean event, double time, int rejectedType) {
     char fastStates_row[4096];
-    sprintf(fastStates_row, "%15.10g %15.10g %15.10g %15.10g", time, gbData->err_slow, gbData->err_int, gbData->err_fast);
+    sprintf(fastStates_row, "%15.10g %2d %15.10g %15.10g %15.10g", time, rejectedType, gbData->err_slow, gbData->err_int, gbData->err_fast);
```
[the remark this answers is not in the document]
Comment (bernhardbachmann) [reply OMCompiler/SimulationRuntime/c/simulation/solver/gbode_util.c:635]: I tried snprintf, but it did not work for me. Need some help with this. Even your implementation of snprintf for the butcher tableau is inconsistent.

File: OMCompiler/SimulationRuntime/c/simulation/solver/gbode_main.c [hunk 7]
```diff
@@ -1313,17 +1317,25 @@ int gbode_birate(DATA *data, threadData_t *threadData, SOLVER_INFO *solverInfo)
         printVector_gbf(LOG_GBODE_V, "e", gbData->errest, nStates, (gbData->timeLeft + gbData->timeRight)/2, gbData->nSlowStates, gbData->slowStatesIdx);
         messageClose(LOG_GBODE_V);
       }
+      if (gbData->ctrl_method != GB_CTRL_CNST && ((gbData->interpolation == GB_INTERPOL_HERMITE_ERRCTRL)  || (gbData->interpolation == GB_DENSE_OUTPUT_ERRCTRL))) {
+        gbData->errValues[0] = fmax(err, gbData->err_int);
+        gbData->stepSize = gbData->lastStepSize * IController(gbData->errValues, gbData->stepSizeValues, gbData->tableau->error_order);
```
[the remark this answers is not in the document]
Comment (bernhardbachmann) [reply OMCompiler/SimulationRuntime/c/simulation/solver/gbode_main.c]: Done

File: OMCompiler/SimulationRuntime/c/simulation/solver/gbode_util.c (hunk 6 above)
[the remark this answers is not in the document]
Comment (AnHeuermann) [reply OMCompiler/SimulationRuntime/c/simulation/solver/gbode_util.c]: Fixed

**Pull request merged**